### PR TITLE
Fix paths in SystemData macro

### PR DESCRIPTION
**Pull request by IsseW**

### CODE REVIEW

**Comment by Imberflur** [comment]
Iirc this is commonly used via a re-export from `specs` so many users won't have the `shred` crate in scope. I think it is worth giving this case some consideration.

**Comment by IsseW** [comment]
> Iirc this is commonly used via a re-export from `specs` so many users won't have the `shred` crate in scope. I think it is worth giving this case some consideration.

Through specs you have `specs::shred` to import in this case. Which I think will be a better thing to have to import than all of `World`, `SystemData` and `ResourceId`. Which could collide with user code.

So maybe if there aren't leading any `::` this could work? I don't think there's a way to solve not having to import anything from specs without creating a seperate proc macro in specs.

**Comment by IsseW** [comment]
Removed leading `::`.

File: src/system.rs [hunk 1]
```diff
@@ -284,6 +284,11 @@ pub trait SystemData<'a> {
     ///
     /// Please note that returning wrong dependencies can lead to a panic.
     fn writes() -> Vec<ResourceId>;
+
+    /// Returns names of retrieved resources.
+    fn resource_names() -> Vec<&'static str> {
+        vec![std::any::type_name::<Self>()]
+    }
```
Comment (Imberflur) [inline src/system.rs:291]: I just noticed that this PR is adding a new method to the SystemData trait. What is the purpose here? Was it intended to be included in this PR?

Comment (IsseW) [reply src/system.rs:291]: Oh huh, I hadn't even noticed that. Must be from some testing a long time ago

**Review by Imberflur — APPROVED** [review]

**Pull request merged**